packages/assets-controllers/src/NftController.ts

```diff
@@ -903,15 +903,17 @@ export class NftController extends BaseControllerV1<NftConfig, NftState> {
       // If the nft is auto-detected we want some valid metadata to be present
       if (
         source === Source.Detected &&
-        // TODO: Replace `any` with type
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        Object.entries(contractInformation).every(([k, v]: [string, any]) => {
+        Object.entries(contractInformation).every(([k, v]) => {
           if (k === 'address') {
             return true; // address will always be present
           }
           // collection will always be an object, we need to check the internal values
           if (k === 'collection') {
-            return v?.name === null && v?.image_url === null;
+            const obj = v as {
+              name: string | null;
+              image_url?: string | null;
+            };
+            return obj?.name === null && obj?.image_url === null;
           }
           return !v;
         })
```

Comment:
The suggestion above is equivalent to the previous code, but based on the type of `contractInformation`:
```ts
Partial<ApiNftContract> & Pick<ApiNftContract, 'address' | 'collection'>
```
(where the `address` and `collection` properties of `ApiNftContract` are non-optional), could the following have been the original intention? All tests are passing with this:

```ts
      if (
        source === Source.Detected &&
        'address' in contractInformation &&
        typeof contractInformation.address === 'string' &&
        'collection' in contractInformation &&
        contractInformation.collection.name === null &&
        'image_url' in contractInformation.collection && 
        contractInformation.collection.image_url === null &&
        Object.entries(contractInformation).every(([key, value]) => {
          return key === 'address' || key === 'collection' || !value;
        })
      ) {
        return nftContracts;
      }
```

Reply:
Usually I lean on the side of honoring the original code, since this is still used in a JavaScript context, so there's a chance that bad data is still being passed through. Your suggestion makes more logical sense though. `(!('address' in contractInformation) || contractInformation.address)` in particular is a very strange check (and similar for `collection`).